src/pages/MatchRoom.js

```diff
@@ -110,28 +39,53 @@ class MatchRoom extends React.Component {
     this.setState({ currentSession });
   };
 
+  getData = async () => {
+    const genreId = this.props.match.params.id;
+    const apiList = await window
+      .fetch(
+        `https://api.themoviedb.org/3/discover/movie?api_key=fcbb1bd6a2b486386efe153e5874f9ee&language=en-US&sort_by=popularity.desc&include_adult=false&include_video=false&page=1&with_genres=${genreId}`
+      )
+      .then((response) => {
+        return response
+          .json()
+          .then((data) => {
+            return data.results;
+          })
+          .catch(() => console.error('api not responding with the list'));
+      });
+    this.setState({ apiList });
+  };
+
+  componentDidMount () {
+    this.getData();
+  }
+
   render () {
     const { user1, user2 } = this.props;
-    return (
-      <div>
-        <HeaderSmall />
-        <Link to='/result'>
-          <h2>Result</h2>
-        </Link>
-        {this.state.currentSession === 'user1' ? (
-          <User1List
-            user1={user1}
-            {...this.state}
-            onHandleSession={this.handleSession}
-            onHandleReject={this.handleReject}
-            onHandleValidate={this.handleValidate}
-            user2={user2}
-          />
-        ) : (
-          <User2List user2={user2} />
-        )}
-      </div>
-    );
+    if (this.state.apiList[0] !== undefined) {
```

Comment:
Je vous conseille de gérer un état de chargement explicite plutôt que de compter sur le fait this.state.apiList[0] est undefined quand les données sont en cours de chargement